examples/controls_example/public/react_controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -0,0 +1,397 @@
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect } from 'react';
+import { BehaviorSubject, combineLatest, debounceTime, filter, skip } from 'rxjs';
+
+import { OptionsListSearchTechnique } from '@kbn/controls-plugin/common/options_list/suggestions_searching';
+import { OptionsListSortingType } from '@kbn/controls-plugin/common/options_list/suggestions_sorting';
+import {
+  OptionsListSuccessResponse,
+  OptionsListSuggestions,
+} from '@kbn/controls-plugin/common/options_list/types';
+import { buildExistsFilter, buildPhraseFilter, buildPhrasesFilter, Filter } from '@kbn/es-query';
+import { useBatchedPublishingSubjects } from '@kbn/presentation-publishing';
+
+import { isValidSearch } from '../../../../common/options_list/is_valid_search';
+import {
+  getSelectionAsFieldType,
+  OptionsListSelection,
+} from '../../../../common/options_list/options_list_selections';
+import { initializeDataControl } from '../initialize_data_control';
+import { DataControlFactory, DataControlServices } from '../types';
+import { OptionsListControl } from './components/options_list_control';
+import { OptionsListEditorOptions } from './components/options_list_editor_options';
+import { OptionsListStrings } from './options_list_strings';
+import {
+  DEFAULT_SEARCH_TECHNIQUE,
+  MIN_OPTIONS_LIST_REQUEST_SIZE,
+  OPTIONS_LIST_CONTROL_TYPE,
+  OPTIONS_LIST_DEFAULT_SORT,
+} from './constants';
+import { fetchAndValidate$ } from './fetch_and_validate';
+import { OptionsListControlContext } from './options_list_context_provider';
+import { OptionsListControlApi, OptionsListControlState } from './types';
+
+export const getOptionsListControlFactory = (
+  services: DataControlServices
+): DataControlFactory<OptionsListControlState, OptionsListControlApi> => {
+  return {
+    type: OPTIONS_LIST_CONTROL_TYPE,
+    getIconType: () => 'editorChecklist',
+    getDisplayName: OptionsListStrings.control.getDisplayName,
+    isFieldCompatible: (field) => {
+      return (
+        !field.spec.scripted &&
+        field.aggregatable &&
+        ['string', 'boolean', 'ip', 'date', 'number'].includes(field.type)
+      );
+    },
+    CustomOptionsComponent: OptionsListEditorOptions,
+    buildControl: async (initialState, buildApi, uuid, controlGroupApi) => {
+      /** Serializable state - i.e. the state that is saved with the control */
+      const searchTechnique$ = new BehaviorSubject<OptionsListSearchTechnique | undefined>(
+        initialState.searchTechnique ?? DEFAULT_SEARCH_TECHNIQUE
+      );
+      const runPastTimeout$ = new BehaviorSubject<boolean | undefined>(initialState.runPastTimeout);
+      const singleSelect$ = new BehaviorSubject<boolean | undefined>(initialState.singleSelect);
+      const selections$ = new BehaviorSubject<OptionsListSelection[] | undefined>(
+        initialState.selectedOptions ?? []
+      );
+      const sort$ = new BehaviorSubject<OptionsListSortingType | undefined>(
```

Comment:
Same comment as above. Can not set default value when creating the observable, instead, set default on read when value is undefined.

Reply:
See https://github.com/elastic/kibana/pull/186655#discussion_r1697305846